lib/spack/spack/environment.py

```diff
@@ -1049,7 +1085,7 @@ def write(self):
     def __enter__(self):
         self._previous_active = _active_environment
         activate(self)
-        return
+        return self
```

Comment:
this seems to be an unrelated change?

Reply:
It's unrelated - I started writing the test using this context manager, then simplified it. Anyhow, it's kind of unexpected to have a context manager that doesn't support the `with <expr> as <name>` syntax. Here `<name>` was bound to `None`.